include/gsl/util

```diff
@@ -72,9 +72,9 @@ private:
 
 // finally() - convenience function to generate a final_action
 template <class F>
-[[nodiscard]] auto finally(F&& f) noexcept
+GSL_NODISCARD auto finally(F&& f) noexcept
 {
-    return final_action<F>{std::forward<F>(f)};
+    return final_action<std::remove_reference_t<F>>{std::forward<F>(f)};
```

Comment:
Final suggestion (sorry I missed this before) we probably also want to strip cv-qualifiers from `F` so the resulting `final_action` doesn't have a member with `const`- or `volatile`-qualified type:
```suggestion
    return final_action<std::remove_cv_t<std::remove_reference_t<F>>>{std::forward<F>(f)};
```

Reply:
Aha, I was wondering why you hadn't suggested `remove_cvref`! :) I figured you had a reason. OK, I'll make this change, and leave it as two separate calls to keep it C++11 and avoid creating a C++20 dependency yet. Thanks.